pkg/controller/pod_test.go

```diff
@@ -616,7 +636,57 @@ var _ = Describe("Dynamic Attachment controller", func() {
 					}).Should(ConsistOf(defaultNet))
 				})
 			})
+			When("an attachment is added with update status failed", func() {
+				JustBeforeEach(func() {
+					var err error
+					pod = updatePodSpec(pod)
+					pod.Annotations[nad.NetworkStatusAnnot] =
+						`[
+							{
+								"name": "default/tiny-net",
+								"interface": "net0",
+								"dns": {}
+							},
+							{
+								"name": "default/tiny-net-2",
+								"interface": "net1",
+								"mac": "02:03:04:05:06:07",
+								"dns": {}
+							}
+						]`
+					_, err = k8sClient.CoreV1().Pods(namespace).UpdateStatus(
+						context.TODO(),
+						updatePodSpec(pod, networkName, networkToAdd, networkToAdd1),
+						metav1.UpdateOptions{})
+					Expect(err).NotTo(HaveOccurred())
+					expectedError := errors.New("someerror")
+                                        k8sClient.PrependReactor("update", "pods", func(_ k8stesting.Action) (bool, runtime.Object, error) {
+                                                return true, nil, expectedError
+                                        })
+				})
+
+				It("an `AddedInterface` and then a `RemovingInterface` event are seen in the event recorded ", func() {
+
+					expectedEventPayload := fmt.Sprintf(
```

Comment:
nit: you could name this something with semantic meaning - `expectedAddInterfaceEvent` for instance.